File: backend/job-application-service/src/main/java/com/jobportal/jobapp/dto/ApplicationRequestDto.java
package com.jobportal.jobapp.dto;

import jakarta.validation.constraints.NotNull;

public class ApplicationRequestDto {

    @NotNull(message = "Job ID is required")
    private Integer jobId;

    @NotNull(message = "Candidate ID is required")
    private Integer candidateId;

    @NotNull(message = "Resume ID is required")
    private Integer resumeId;

    public ApplicationRequestDto() {
    }

    public Integer getJobId() {
        return jobId;
    }

    public void setJobId(Integer jobId) {
        this.jobId = jobId;
    }

    public Integer getCandidateId() {
        return candidateId;
    }

    public void setCandidateId(Integer candidateId) {
        this.candidateId = candidateId;
    }

    public Integer getResumeId() {
        return resumeId;
    }

    public void setResumeId(Integer resumeId) {
        this.resumeId = resumeId;
    }
}

File: backend-dotnet/Services/JobPortal.JobApplicationService/DTOs/ApplicationRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace JobPortal.JobApplications.DTOs;

public class ApplicationRequestDto
{
    [Required(ErrorMessage = "Job ID is required")]
    public int? JobId { get; set; }

    [Required(ErrorMessage = "Candidate ID is required")]
    public int? CandidateId { get; set; }

    [Required(ErrorMessage = "Resume ID is required")]
    public int? ResumeId { get; set; }
}
